add-on/chrome/test/mochitest/browser_copypanel.js

```diff
@@ -0,0 +1,114 @@
+  histogram.clear();
+  LoopUI.removeCopyPanel();
+  gURLBar.value = "http://" + domain + "/";
+
+  LoopUI.addCopyPanel();
+  gURLBar.focus();
+  gURLBar.select();
+  goDoCommand("cmd_copy");
+
+  let iframe = document.getElementById("loop-copy-panel-iframe");
+  return new Promise(resolve => {
+    iframe.addEventListener("DOMContentLoaded", () => {
+      iframe.parentNode.addEventListener("popuphidden", () => resolve(histogram));
+      setTimeout(() => onIframe(iframe));
+    });
+  });
+}
+
+// Show the copy panel on location bar copy.
+add_task(function* test_copy_panel_shown() {
+  let histogram = yield testClick("some.site", iframe => {
+    iframe.parentNode.hidePopup();
+  });
+
+  Assert.notEqual(document.getElementById("loop-copy-notification-panel"), null, "copy panel exists on copy");
+  Assert.strictEqual(histogram.snapshot().counts[gConstants.COPY_PANEL.SHOWN], 1, "triggered telemetry count for showing");
+});
+
+// Click the accept button without checkbox.
+add_task(function* test_click_yes_again() {
+  let histogram = yield testClick("yes.again", iframe => {
+    iframe.contentDocument.querySelector(".copy-button:last-child").click();
+  });
+
+  Assert.notEqual(document.getElementById("loop-copy-notification-panel"), null, "copy panel still exists");
+  Assert.strictEqual(histogram.snapshot().counts[gConstants.COPY_PANEL.YES_AGAIN], 1, "triggered telemetry count for yes again");
+});
+
+// Click the accept button with checkbox.
+add_task(function* test_click_yes_never() {
+  let histogram = yield testClick("yes.never", iframe => {
+    iframe.contentDocument.querySelector(".copy-toggle-label").click();
+    iframe.contentDocument.querySelector(".copy-button:last-child").click();
+  });
+
+  Assert.equal(document.getElementById("loop-copy-notification-panel"), null, "copy panel removed");
+  Assert.strictEqual(histogram.snapshot().counts[gConstants.COPY_PANEL.YES_NEVER], 1, "triggered telemetry count for yes never");
+});
+
+// Click the cancel button without checkbox.
+add_task(function* test_click_no_again() {
+  let histogram = yield testClick("no.again", iframe => {
+    iframe.contentDocument.querySelector(".copy-button").click();
+  });
+
+  Assert.notEqual(document.getElementById("loop-copy-notification-panel"), null, "copy panel still exists");
+  Assert.strictEqual(histogram.snapshot().counts[gConstants.COPY_PANEL.NO_AGAIN], 1, "triggered telemetry count for no again");
+});
+
+// Click the cancel button with checkbox.
+add_task(function* test_click_no_never() {
+  let histogram = yield testClick("no.never", iframe => {
+    iframe.contentDocument.querySelector(".copy-toggle-label").click();
+    iframe.contentDocument.querySelector(".copy-button").click();
+  });
+
+  Assert.equal(document.getElementById("loop-copy-notification-panel"), null, "copy panel removed");
+  Assert.strictEqual(histogram.snapshot().counts[gConstants.COPY_PANEL.NO_NEVER], 1, "triggered telemetry count for no never");
+});
```

Comment:
There's nothing here that seems to test that the semantics of the _NEVER _AGAIN works.   It seems like that shouldn't be too easy, though possibly even easier to do in a unit-style test rather than by driving the UI in an integration test.  Either strategy could be added to this file, I'd say.